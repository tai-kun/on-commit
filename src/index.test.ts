import * as cp from "node:child_process"
import * as path from "node:path"
import * as process from "node:process"

import { test, expect } from "@jest/globals"

test("test", () => {
  const common = {
    author: {
      email: "foo@bar.baz",
      name: "foo",
      username: "foo"
    },
    committer: {
      email: "foo@bar.baz",
      name: "foo",
      username: "foo"
    },
    id: "9a6747fc6259aa374ab4e1bb03074b6ec672cf99",
    tree_id: "553ae7da92f5505a92bbb8c9d47be76ab9f65bc2",
    url: "https://github.com/foo/bar/commit/9a6747fc6259aa374ab4e1bb03074b6ec672cf99"
  }
  process.env["INPUT_COMMITS"] = JSON.stringify([
    {
      ...common,
      message: "ðŸš€ ãƒ“ãƒ«ãƒ‰"
    },
    {
      ...common,
      message: "ðŸ”– v2.2.0"
    }
  ])

  const np = process.execPath
  const ip = path.join(__dirname, "..", "dist", "index.cjs")

  expect(cp.execFileSync(np, [ip], { env: process.env }).toString())
    .toMatchSnapshot()
})
