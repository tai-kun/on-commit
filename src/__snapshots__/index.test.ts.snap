// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test 1`] = `
"::debug::regex: /v(?<version>(?<versionMajor>(0|[1-9]\\d*))\\.(?<versionMinor>(0|[1-9]\\d*))\\.(?<versionPatch>(0|[1-9]\\d*))(?<versionTag>(-(?<versionTagName>\\w+)\\.(?<versionTagNumber>(0|[1-9]\\d*))))?)\\s*$/
::debug::[%0A  {%0A    "author": {%0A      "email": "foo@bar.baz",%0A      "name": "foo",%0A      "username": "foo"%0A    },%0A    "committer": {%0A      "email": "foo@bar.baz",%0A      "name": "foo",%0A      "username": "foo"%0A    },%0A    "id": "9a6747fc6259aa374ab4e1bb03074b6ec672cf99",%0A    "message": "ðŸš€ ãƒ“ãƒ«ãƒ‰",%0A    "tree_id": "553ae7da92f5505a92bbb8c9d47be76ab9f65bc2",%0A    "url": "https://github.com/foo/bar/commit/9a6747fc6259aa374ab4e1bb03074b6ec672cf99"%0A  },%0A  {%0A    "author": {%0A      "email": "foo@bar.baz",%0A      "name": "foo",%0A      "username": "foo"%0A    },%0A    "committer": {%0A      "email": "foo@bar.baz",%0A      "name": "foo",%0A      "username": "foo"%0A    },%0A    "id": "9a6747fc6259aa374ab4e1bb03074b6ec672cf99",%0A    "message": "ðŸ”– v2.2.0",%0A    "tree_id": "553ae7da92f5505a92bbb8c9d47be76ab9f65bc2",%0A    "url": "https://github.com/foo/bar/commit/9a6747fc6259aa374ab4e1bb03074b6ec672cf99"%0A  }%0A]
::debug::message: ðŸ”– v2.2.0
::debug::RegExpExecArray.groups: ["v2.2.0","2.2.0","2","2","2","2","0","0",null,null,null,null,null]
::debug::RegExpExecArray.groups: {"version":"2.2.0","versionMajor":"2","versionMinor":"2","versionPatch":"0"}
::debug::{"version":"2.2.0","version-major":"2","version-minor":"2","version-patch":"0","version-tag":"","version-tag-name":"","version-tag-number":""}

::set-output name=version::2.2.0

::set-output name=version-major::2

::set-output name=version-minor::2

::set-output name=version-patch::0

::set-output name=version-tag::

::set-output name=version-tag-name::

::set-output name=version-tag-number::

::set-output name=commit-author-email::foo@bar.baz

::set-output name=commit-author-name::foo

::set-output name=commit-author-username::foo

::set-output name=commit-committer-email::foo@bar.baz

::set-output name=commit-committer-name::foo

::set-output name=commit-id::9a6747fc6259aa374ab4e1bb03074b6ec672cf99

::set-output name=commit-message::ðŸ”– v2.2.0

::set-output name=commit-tree-id::553ae7da92f5505a92bbb8c9d47be76ab9f65bc2

::set-output name=commit-url::https://github.com/foo/bar/commit/9a6747fc6259aa374ab4e1bb03074b6ec672cf99

::set-output name=matched::true
"
`;
